compiler/msgs.nim

```diff
@@ -734,20 +734,19 @@ proc outWriteln*(s: string) =
     flushFile(stdout)
 
 proc msgWriteln*(s: string) =
-  ## Writes to stdout. If --stdout option is given, writes to stderr instead.
+  ## Writes to stderr. If --stdout option is given, writes to stdout instead.
 
   #if gCmd == cmdIdeTools and optCDebug notin gGlobalOptions: return
 
   if not isNil(writelnHook):
     writelnHook(s)
   elif optStdout in gGlobalOptions:
-    if eStdErr in errorOutputs:
-      writeLine(stderr, s)
-      flushFile(stderr)
-  else:
     if eStdOut in errorOutputs:
       writeLine(stdout, s)
       flushFile(stdout)
+  else:
+    if eStdErr in errorOutputs:
+      writeLine(stderr, s)
```

Comment:
@Araq My tests were indeed incomplete. When using MSVCRT and piping `stderr` to file it becomes fully-buffered, so Microsoft is indeed pissing on C standard.
Added back `flushFile(stderr)` when targeting `windows`.